mlrun/api/db/sqldb/db.py

```diff
@@ -1330,11 +1330,22 @@ def store_feature_set(
         _, _, existing_feature_set = self._get_record_by_name_tag_and_uid(
             session, FeatureSet, project, name, tag, uid
         )
+
+        feature_set_dict = feature_set.dict()
+
         if not existing_feature_set:
+            # Check if this is a re-tag of existing object - search by uid only
+            uid = fill_object_hash(feature_set_dict, "uid", tag)
+            _, _, existing_feature_set = self._get_record_by_name_tag_and_uid(
+                session, FeatureSet, project, name, None, uid
+            )
+            if existing_feature_set:
+                self.tag_objects_v2(session, [existing_feature_set], project, tag)
+                return uid
+
```

Comment:
I feel like this is an unneeded "branching" of the logic that isn't really needed
The store flow should be idempotent, So nothing wrong in doing an the extra store to the object itself (it wasn't change anyways) I don't see much value from saving it
I would simplify the code and simply get only by uid back in the first time you do the  `_get_record_by_name_tag_and_uid`
But I might be missing something, am I ?

Reply:
It's a different logic - the `_get_record_by_name_tag_and_uid`, when called with a tag, just looks for the object with that name and tag and returns the `uid`. This means that regardless of the object's content, if an object instance with that name and tag exist, they will be returned. Then they can be overridden by the rest of the method. However, if we follow that logic here, then actually storing the object will fail, due to DB constraints (the insert itself will fail).
To follow your suggestion, branching will need to be introduced in `create_feature_set` to check for existence of object except for tag. It makes the `create_feature_set` function more complicated and may impact pure insert (`POST`) flows as well. OTOH, putting the code here focuses it around the re-store-with-different-tag, which IMO is cleaner.